src/internal.c

```diff
@@ -13609,6 +13609,7 @@ int ProcessPeerCerts(WOLFSSL* ssl, byte* input, word32* inOutIdx,
     (defined(WOLFSSL_CERT_REQ) || defined(WOLFSSL_CERT_EXT)) && \
     !defined(NO_FILESYSTEM) && !defined(NO_WOLFSSL_DIR)
                     if (ret == ASN_NO_SIGNER_E || ret == ASN_SELF_SIGNED_E) {
+                        args->lastErr = ret; /* save error from last time */
```

Comment:
This all looks good. Did you consider using a local `int lastErr` vs the shared args one? I don't think the call to `LoadCertByIssuer` can return a `WC_PENDING_E` or `OCSP_WANT_READ`. Wondering if the use of `args->lastErr` could have other repercussions.

Reply:
That's a good point, I don't want to potentially mix together different groups of return errors.

I'll use local `int lastErr = ret;`.

Reply:
note: we discussed adding enhanced testing for this in a later PR.